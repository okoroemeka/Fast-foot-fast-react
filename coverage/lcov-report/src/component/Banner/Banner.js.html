<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/component/Banner/Banner.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/component/Banner</a> Banner.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import { Link } from 'react-router-dom';
// import "./Banner.css";
&nbsp;
const Banner = () =&gt; (
  &lt;div className="row hero-container"&gt;
    &lt;div className="col-12 hero" id="hero"&gt;
      &lt;div className="row overlay-container"&gt;
        &lt;div className="col-12" id="hero-overlay"&gt;
          &lt;div className="row caption-container"&gt;
            &lt;div className="col-10 caption"&gt;
              &lt;div className="row"&gt;
                &lt;div className="col-12"&gt;
                  &lt;p id="home-page-text"&gt;
                    Home of beautiful Intercontinental cuisins and fast
                    delivery.
                  &lt;/p&gt;
                  &lt;p id="home-page-signup-text"&gt;Sign up and lets feed you&lt;/p&gt;
                &lt;/div&gt;
              &lt;/div&gt;
              &lt;div className="row landing-page-button-container"&gt;
                &lt;div className="col-12 home-buttons"&gt;
                  &lt;Link to="signup" id="landingPage-sign-up"&gt;
                    Sign up
                  &lt;/Link&gt;
                  &lt;Link to="login" id="landingPage-logIn-button"&gt;
                    Log In
                  &lt;/Link&gt;
                &lt;/div&gt;
              &lt;/div&gt;
            &lt;/div&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/div&gt;
);
&nbsp;
export default Banner;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Feb 04 2019 20:20:46 GMT+0100 (West Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
