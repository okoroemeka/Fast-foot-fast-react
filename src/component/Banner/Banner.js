import React, { Fragment } from "react";
import { Link } from "react-router-dom";

const logInStatus = JSON.parse(window.localStorage.getItem("isLoggedIn"));
const Banner = () => {
  const bannerButton = !logInStatus ? (
    <Fragment>
      <Link to="signup" id="landingPage-sign-up">
        Sign up
      </Link>
      <Link to="login" id="landingPage-logIn-button">
        Log In
      </Link>
    </Fragment>
  ) : null;
  return (
    <div className="row hero-container">
      <div className="col-12 hero" id="hero">
        <div className="row overlay-container">
          <div className="col-12" id="hero-overlay">
            <div className="row caption-container">
              <div className="col-10 caption">
                <div className="row">
                  <div className="col-12">
                    <p id="home-page-text">
                      Home of beautiful Intercontinental cuisins and fast
                      delivery.
                    </p>
                    <p id="home-page-signup-text">Sign up and lets feed you</p>
                  </div>
                </div>
                <div className="row landing-page-button-container">
                  <div className="col-12 home-buttons" >
                    {bannerButton}
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Banner;
