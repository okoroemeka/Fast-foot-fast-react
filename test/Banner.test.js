import React from 'react';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import {Router} from 'react-router-dom'
import {createMemoryHistory} from 'history'
import { render, cleanup } from 'react-testing-library';
import Banner from '../src/component/Banner/Banner';


afterEach(cleanup);

describe('<Banner />', () => {
  const history = createMemoryHistory({initialEntries: ['/']})
  test('render correctly', () => {
    const  banner  = render(
      <Router history={history}>
        <Banner />
      </Router>,
    );
    expect(banner).toMatchSnapshot();
  });
});