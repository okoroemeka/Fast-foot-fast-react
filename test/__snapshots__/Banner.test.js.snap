// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row hero-container"
      >
        <div
          class="col-12 hero"
          id="hero"
        >
          <div
            class="row overlay-container"
          >
            <div
              class="col-12"
              id="hero-overlay"
            >
              <div
                class="row caption-container"
              >
                <div
                  class="col-10 caption"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-12"
                    >
                      <p
                        id="home-page-text"
                      >
                        Home of beautiful Intercontinental cuisins and fast delivery.
                      </p>
                      <p
                        id="home-page-signup-text"
                      >
                        Sign up and lets feed you
                      </p>
                    </div>
                  </div>
                  <div
                    class="row landing-page-button-container"
                  >
                    <div
                      class="col-12 home-buttons"
                    >
                      <a
                        href="/signup"
                        id="landingPage-sign-up"
                      >
                        Sign up
                      </a>
                      <a
                        href="/login"
                        id="landingPage-logIn-button"
                      >
                        Log In
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row hero-container"
    >
      <div
        class="col-12 hero"
        id="hero"
      >
        <div
          class="row overlay-container"
        >
          <div
            class="col-12"
            id="hero-overlay"
          >
            <div
              class="row caption-container"
            >
              <div
                class="col-10 caption"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-12"
                  >
                    <p
                      id="home-page-text"
                    >
                      Home of beautiful Intercontinental cuisins and fast delivery.
                    </p>
                    <p
                      id="home-page-signup-text"
                    >
                      Sign up and lets feed you
                    </p>
                  </div>
                </div>
                <div
                  class="row landing-page-button-container"
                >
                  <div
                    class="col-12 home-buttons"
                  >
                    <a
                      href="/signup"
                      id="landingPage-sign-up"
                    >
                      Sign up
                    </a>
                    <a
                      href="/login"
                      id="landingPage-logIn-button"
                    >
                      Log In
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
